js/outputter.js

```diff
@@ -0,0 +1,568 @@
+        cite.innerText = this.config.sampleText["url"];
+        cite.contentEditable = true;
+    }
+    target.appendChild( cite );
+};
+
+/**
+ * creates the meta description elements in the snippetPreview and appends to target
+ * @param target
+ */
+AnalyzeLoader.prototype.createSnippetPreviewMeta = function ( target ){
+    var meta = document.createElement( "span" );
+    meta.className = "desc";
+    meta.id = "snippet_meta";
+    if( this.config.inputType === "both" || this.config.inputType === "inline" ){
+        meta.contentEditable = true;
+        meta.innerText = this.config.sampleText["meta"];
+    }
+    target.appendChild( meta );
+};
+
+/**
+ * defines the target element to be used for the output on the page
+ */
+AnalyzeLoader.prototype.defineElements = function() {
+    this.target = document.getElementById( this.config.targets.output );
+};
+
+/**
+ * gets the values from the inputfields. The values from these fields are used as input for the analyzer.
+ */
+AnalyzeLoader.prototype.getInput = function() {
+    this.inputs.textString = document.getElementById( "textInput" ).value;
+    this.inputs.keyword = document.getElementById( "keywordInput" ).value;
+    this.inputs.meta = document.getElementById( "metaInput" ).value;
+    this.inputs.url = document.getElementById( "urlInput" ).value;
+    this.inputs.pageTitle = document.getElementById( "titleInput" ).value;
+};
+
+/**
+ * binds the events to the generated inputs. Binds events on the snippetinputs if editable
+ */
+AnalyzeLoader.prototype.bindEvent = function() {
+    this.bindInputEvent();
+    if( this.config.inputType === "both" || this.config.inputType === "inline") {
+        this.bindSnippetEvents();
+    }
+};
+
+/**
+ * binds the analyzeTimer function to the input of the targetElement on the page.
+ */
+AnalyzeLoader.prototype.bindInputEvent = function() {
+    var elem = document.getElementById( this.config.elementTarget );
+    elem.__refObj = this;
+    elem.addEventListener( "input", this.analyzeTimer );
+};
+
+/**
+ * binds the reloadSnippetText function to the blur of the snippet inputs.
+ */
+AnalyzeLoader.prototype.bindSnippetEvents = function() {
+    var snippetElem = document.getElementById(this.config.targets.snippet);
+    snippetElem.refObj = this;
+    snippetElem.addEventListener("input", this.callBackSnippetData);
+    var elems = ["meta", "cite", "title"];
+    for (var i = 0; i < elems.length; i++) {
+        var targetElement = document.getElementById("snippet_" + elems[i]);
+        targetElement.refObj = this;
+        targetElement.addEventListener( "blur", this.reloadSnippetText );
+    }
+};
+
+/**
+ * callBackSnippetData is bound on the inputs of the snippetPreview, to update the inputs when entering text
+ * in the snippetPreview.
+ */
+AnalyzeLoader.prototype.callBackSnippetData = function() {
+    document.getElementById( "titleInput" ).value = document.getElementById( "snippet_title" ).innerText;
```

Comment:
Maybe you set this as properties somewhere and access the elements by properties, this might be more readable than using `document.getElementById()`.